src/accelerate/state.py

```diff
@@ -426,7 +427,7 @@ def _goes_first(self, is_main: bool):
             self.wait_for_everyone()
 
     @contextmanager
-    def split_between_processes(self, inputs: list | tuple | dict | torch.Tensor, apply_padding: bool = False):
+    def split_between_processes(self, inputs: list | tuple | dict | torch.Tensor | Dataset, apply_padding: bool = False):
```

Comment:
Use `"Dataset"` instead. Or `"datasets.Dataset"`. 

If that doesn't work let's just add a snippet in the doc of "can also split `datasets.Dataset`s"